Source/MatcherFactory.cs

```diff
@@ -95,22 +95,21 @@ public static IMatcher CreateMatcher(Expression expression, bool isParams)
 				}
 
 				var attr = call.Method.GetCustomAttribute<AdvancedMatcherAttribute>(true);
-#pragma warning disable 618
-				var staticMatcherMethodAttr = call.Method.GetCustomAttribute<MatcherAttribute>(true);
-#pragma warning restore 618
 
 				if (attr != null)
 				{
 					var matcher = attr.CreateMatcher();
 					matcher.Initialize(originalExpression);
 					return matcher;
 				}
-				else if (staticMatcherMethodAttr != null)
+#pragma warning disable 618
+				else if (call.Method.IsDefined(typeof(MatcherAttribute), true))
```

Comment:
Minor optimization here: We don't need the actual attribute (`GetCustomAttribute`; above), we just need to know whether it is there or not (`IsDefined`). This call is about twice as fast.